### ISSUE TITLE
$page.error not typed as HttpError

### ISSUE THREAD

**Issue by austins** [sveltejs/kit#5970]
### Describe the bug

In SvelteKit `1.0.0-next.415`, `$page.error` isn't typed as HttpError (which is not exported publicly).

### Reproduction

If you try to output `$page.error.message` in `+error.svelte` you'll get an `Unsafe member access .message on an any value. eslint(@typescript-eslint/no-unsafe-member-access)` error.

```
<script lang="ts">
    import { page } from "$app/stores";
</script>

<h1>{$page.status}: {$page.error.message}</h1>
```

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (24) x64 AMD Ryzen 9 3900X 12-Core Processor
    Memory: 2.33 GB / 15.93 GB
  Binaries:
    Node: 16.16.0 - ~\scoop\apps\nodejs-lts\current\node.EXE
    npm: 8.15.1 - ~\scoop\apps\nodejs-lts\current\bin\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (104.0.1293.47)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    @sveltejs/adapter-node: ^1.0.0-next.86 => 1.0.0-next.86
    @sveltejs/kit: ^1.0.0-next.415 => 1.0.0-next.415
    @sveltejs/vite-plugin-svelte: ^1.0.1 => 1.0.1
    svelte: ^3.49.0 => 3.49.0
    vite: ^3.0.8 => 3.0.8
```

### Severity

annoyance

**Comment by dummdidumm** [comment]
This is correct, because this can be something else than `HttpError`, too (an `Error` for example, though that one has `message`, too). Not sure if we can do something here to appease the common case where message exists.

**Comment by austins** [comment]
For now, we can cast the type like so:
```
    $: error = $page.error as Error;
```

**Comment by Rich-Harris** [comment]
Since errors have to be converted to POJOs to cross the serialization boundary between server and client, their type is really something like

```ts
{ status: number, message: string } |
{ name: string, message: string, [key: string]: any }
```

I feel like we could probably make this work?

**Comment by dummdidumm** [comment]
Closed via #6586
